app/controllers/publishers/uphold_controller.rb

```diff
@@ -0,0 +1,114 @@
+module Publishers
+  class UpholdController < ApplicationController
+    # TODO Refactor Uphold Status to not actually need helper
+    # Traditional usage of helpers should really only be for views
+    include PublishersHelper
+
+    def uphold_status
+      publisher = current_publisher
+      respond_to do |format|
+        format.json do
+          render(json: {
+            uphold_status: publisher.uphold_connection&.uphold_status.to_s,
+            uphold_status_summary: uphold_status_summary(publisher),
+            uphold_status_description: uphold_status_description(publisher),
+            uphold_status_class: uphold_status_class(publisher),
+          }, status: 200)
+        end
+      end
+    end
+
+    def confirm_default_currency_params
+      params.require(:publisher).permit(:default_currency)
+    end
+
+    # Records default currency preference
+    # If user does not have Uphold's `cards:write` scope, we redirect to Uphold to get authorization
+    # Card creation is done in #home
+    def confirm_default_currency
+      # TODO Consider refactoring this
+      uphold_connection = current_publisher.uphold_connection
+
+      uphold_connection.update(confirm_default_currency_params.merge(default_currency_confirmed_at: Time.now))
+
+      if uphold_connection.can_create_uphold_cards?
+        uphold_connection.create_uphold_card_for_default_currency
```

Comment:
Would this run for users who disconnected and re-connected?

Reply:
I believe so, this might cause a 401 error if they are confirming a default currency. However from the UI we don't allow the user to change their default currency when they're in the "Reauthorize state.